src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -183,8 +183,17 @@ cpdef object compute_network(int nsteps, list reaches, dict connections,
 
     # for ts in flowveldepth:
         # print(f"{list(ts)}")
+
+    cdef set fill_index_mask = set()
+    # fill_index_mask is filled in the explicit loop below, which is
+    # identical to the following comprehension. But we use the explicit loop, because it
+    # is more transparent (and therefore optimizable) to cython.
+    # cdef set fill_index_mask = set([upstream_results[upstream_tw_id]["position_index"] for upstream_tw_id in upstream_results])
+    #print(f"{fill_index_mask}")
+
     for upstream_tw_id in upstream_results:
         fill_index = upstream_results[upstream_tw_id]["position_index"]
+        fill_index_mask.add(upstream_results[upstream_tw_id]["position_index"])
```

Comment:
@groutr ... and this is where the fill_index_mask is generated. 

you were observing that in this loop, the subloop should be replaceable by a slice of some kind... like `flowveldepth[flow_index, :] = upstream_results[upstream_tw_id]["results"]`

But that returns this error: 
```
/shared/t-route-jsh/src/python_routing_v02/fast_reach/mc_reach.pyx in mc_reach.compute_network()
    200         # for idx, val in enumerate(upstream_results[upstream_tw_id]["results"]):
    201         #     flowveldepth[fill_index][idx] = val
--> 202         flowveldepth[fill_index] = upstream_results[upstream_tw_id]["results"]
    203         # print(f"Now filled, it contains:")
    204         # print(f"{list(flowveldepth[fill_index])}")

TypeError: only size-1 arrays can be converted to Python scalars
```